support-models/src/main/scala/com/gu/support/workers/states/SendThankYouEmailState.scala

```diff
@@ -2,36 +2,139 @@ package com.gu.support.workers.states
 
 import java.util.UUID
 
-import com.gu.support.encoding.CustomCodecs.{decodeLocalTime, encodeLocalTime}
+import com.gu.support.encoding.Codec
+import com.gu.support.encoding.Codec.deriveCodec
+import com.gu.support.encoding.CustomCodecs._
 import com.gu.support.promotions.PromoCode
-import com.gu.support.redemptions.RedemptionData
+import com.gu.support.workers.GiftRecipient.{DigitalSubscriptionGiftRecipient, WeeklyGiftRecipient}
+import com.gu.support.workers.states.ProductTypeCreated.DigitalSubscriptionCreated._
+import com.gu.support.workers.states.ProductTypeCreated.{ContributionCreated, GuardianWeeklyCreated, PaperCreated}
 import com.gu.support.workers.{PaymentMethod, SalesforceContactRecord, User, _}
+import io.circe.syntax.EncoderOps
+import io.circe.{Decoder, Encoder}
 import org.joda.time.LocalDate
 
 case class SendThankYouEmailState(
   requestId: UUID,
   user: User,
-  giftPurchase: Option[GiftPurchase],
-  product: ProductType,
   analyticsInfo: AnalyticsInfo,
-  paymentOrRedemptionData: Either[PaymentMethodWithSchedule, RedemptionData],
-  firstDeliveryDate: Option[LocalDate],
-  promoCode: Option[PromoCode],
+  productTypeCreated: ProductTypeCreated,
   salesForceContact: SalesforceContactRecord,
-  accountNumber: String,
-  subscriptionNumber: String,
   acquisitionData: Option[AcquisitionData]
 ) extends SendAcquisitionEventState
 
+case class PurchaseInfo(
+  paymentMethod: PaymentMethod,
+  paymentSchedule: PaymentSchedule,
+  promoCode: Option[PromoCode],
+  accountNumber: String,
+  subscriptionNumber: String,
+)
```

Comment:
just grouping a bunch of things you always have in the state for a purchase (in the existing model).  Although it probably should be refined later as necessary e.g. contribution doesn't have a promo code, or care about the payment schedule.